src/services/machines/funkeC2ShareMachineService.ts

```diff
@@ -201,20 +209,32 @@ export const siopSendResponse = async (
 
   const pd = authorizationRequestData.presentationDefinitions?.[0].definition
   const pex: PEX = new PEX({hasher: generateDigest});
-  sharedCredential.forEach(credential => {
-    let data
+  for (const credential of Array.from(sharedCredentials.values())) {
```

Comment:
This for is not a callback anymore. Please double-check this loop. Below it has

return store.dispatch<any>
It looks like inside this loop I think, but the function is supposed to return response. (It's difficult to determine all the closing braces in here.)

Reply:
fixed